web/html/src/manager/maintenance/calendar/web-calendar.tsx

```diff
@@ -0,0 +1,262 @@
+  const onDateClick = (info) => {
+    setCurrentDate(getApi().currentDataManager.data.currentDate);
+    getApi().changeView('timeGridDay', info.date);
+    if (needsUpdate(info.date)) {
+      getEvents("current");
+    }
+  };
+
+  // Get the date used to fetch events based on the operation
+  const getDate = (operation) => {
+    switch (operation) {
+      case "next":
+        return Date.parse(getApi().currentDataManager.state.dateProfile.currentRange.end);
+      case "back":
+        // currentRange.start returns the first day of the displayed month, but we want to end up in
+        // the previous month so we subtract one day in this case
+        return Date.parse(getApi().currentDataManager.state.dateProfile.currentRange.start) - 24 * 60 * 60 * 1000;
+      case "current":
+        return Date.parse(getApi().currentDataManager.data.currentDate);
+      default:
+        return Date.parse(getApi().currentDataManager.data.dateProfileGenerator.nowDate);
+    }
+  };
+
+  const navigateTo = (operation) => {
+    switch (operation) {
+      case "next":
+        getApi().next();
+        break;
+      case "back":
+        getApi().prev();
+        break;
+      case "current":
+        getApi().gotoDate(getApi().currentDataManager.data.currentDate);
+        break;
+      default:
+        getApi().today();
+    }
+  }
+
+  const needsUpdate = (date) => {
+    return (moment(date).month() !== moment(currentDate).month() ||
+      moment(date).year() !== moment(currentDate).year());
+  };
+
+  const skipToNextEvent = (date) => {
+    // We are only interested in events from the same month that are later or equal to 'date'
+    const filteredEvents = events.filter(event => (
+      moment.parseZone(event.start).month() === moment(currentDate).month() &&
+      moment.parseZone(event.start).date() >= moment(date).date())
+    );
+    // Only perform action if there is an event to go to
+    filteredEvents.length > 0 && getApi().gotoDate(filteredEvents[0].start);
+    // Return true if there's an event false otherwise
+    return filteredEvents.length > 0;
+  }
+
+  const skipToPrevEvent = (date) => {
+    // We are only interested in events from the same month that are earlier or equal to 'date'
+    const filteredEvents = events.filter(event => (
+      moment.parseZone(event.start).month() === moment(currentDate).month() &&
+      moment.parseZone(event.start).date() <= moment(date).date())
+    );
+    // Only perform action if there is an event to go to
+    filteredEvents.length > 0 && getApi().gotoDate(filteredEvents[filteredEvents.length - 1].start);
+    // Return true if there's an event false otherwise
+    return filteredEvents.length > 0;
+  }
+
+  return (
+    <div>
+      <FullCalendar
+        ref={calendarRef}
+        timeZone={""} // Prevent FullCalendar from using the browsers set timezone
+        locales={allLocales}
+        locale={window.preferredLocale.replace("_", "-")}
```

Comment:
Would appreciate a comment here to be clear about what format we're going to and from, right now it's hard to say whether this is safe or not without going to the backend code.

Reply:
Sure I can add a comment. At the moment we use all the locales with `_` on the backend while Fullcalendar expects those with `-` instead